tests/server/activity-event.js

```diff
@@ -0,0 +1,296 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+define([
+  'intern!object',
+  'intern/chai!assert',
+  'intern/dojo/node!../../server/lib/activity-event',
+  'intern/dojo/node!path',
+  'intern/dojo/node!sinon',
+], function (registerSuite, assert, activityEvent, path, sinon) {
+  var write;
+
+  registerSuite({
+    name: 'activity-event',
+
+    'interface is correct': function () {
+      assert.isFunction(activityEvent);
+      assert.lengthOf(activityEvent, 3);
+    },
+
+    'call activityEvent with service and entrypoint, without client_id and entryPoint': {
+      setup: function () {
+        write = process.stderr.write;
+        process.stderr.write = sinon.spy();
+        return activityEvent('mock event', { a: 'b', c: 'd' }, {
```

Comment:
Much cleaner to return the promise and let the test runner deal with the asynchronous behavior! :+1: